### PR TITLE
table view of users

### PR DESCRIPTION
**Pull request by Ewiseman**
Kind of stuck...and it feels like I should know how to do this.

I want to add another index view showing my users in a table format.  I've added the view template, but I am having a hard time figuring out the proper route.  I'm either making this more complex than it is or this is something I've not come across.  

I've been at this for a few hours now and throwing in the towel for now.  If you could point me in the right direction I would greatly appreciate it. 

Thanks!

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -19,6 +19,16 @@
         </div>
 
         <div class="form-group">
+          <%= f.label :time_zone, "* Time Zone" %>
+          <%= f.select :time_zone, ["Eastern", "Central", "Mountain", "Pacific", "Alaskan", "Hawaiian", "Asian", "European", "Other" ], {prompt: ""}, {class: "form-control"} %>
+        </div>
+
+        <div class="form-group">
+          <%= f.label :state, "State:" %>
+          <%= f.text_field :state, class: 'form-control', placeholder: "ex. Colorado", autofocus: true %>
```
Comment (eliotsykes) [inline app/views/devise/registrations/edit.html.erb]: Only one field per page should have `autofocus: true`. Read up on the option if you're unclear on its purpose.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -10,12 +10,10 @@ def update
     
   end
 
+
   def index
     @users = User.all
-    @admin_users = User.where(role: 'admin')
```
Comment (eliotsykes) [inline app/controllers/users_controller.rb]: These deleted variables (from lines 15-18) will no longer be available to the view template for `GET /users`.

**Comment by eliotsykes** [comment]
Hey @Ewiseman,

What's your ideal desired behaviour for these 2 URL paths?

- `/users`
- `/users/team_table`

Should they both exist and serve entirely different views? Does only one of them need to exist?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Ewiseman) [inline app/controllers/users_controller.rb]: Yeah, I think that is okay.  I was going to use them to organize the user by role on the user#index page but I ended up doing with some **if** logic.

File: app/views/devise/registrations/edit.html.erb (hunk 1 above)
Comment (Ewiseman) [inline app/views/devise/registrations/edit.html.erb]: Thanks.  One of those things I should have probably researched before I used.  Got it now.

**Comment by Ewiseman** [comment]
Ideally I would like to have the current **user#index page (/user)** live exactly as it does now.  Then I would like to have a link within the **user#index** page that says "see users in table form" and that takes them to another page (users/team_table) that has a table of the users and their corresponding information.   

Does that make sense?  Essentially I want two user#index pages - avatar layout and table layout.

**Comment by eliotsykes** [comment]
@Ewiseman with the current routes.rb, here's the order that the user-related routes appear in:

```ruby
  resources :users, only: [:update, :index, :show]
  ...
  get 'users/team_table'
```

The order of routes in `routes.rb` is important. The nearer the top of the file that a route is defined, the higher priority it has when Rails attempts to match an incoming request URL with the right controller+action combination.

As it stands, the `/users/team_table` route will never be reached as its snagged first by the higher priority `resources` `:show` route.

The `resources :users` `:show` route matches URLs with the format `/users/:id`, where that `:id` dynamic segment can be *any* string (its not constrained to id numbers only although the name suggests otherwise). So all of these URLs would end up being sent to `UsersController#show`:

- `/users/5`
- `/users/532487346`
- `/users/any-old-nonsense-waefsk23237w-werwa`
- `/users/x`
- and most importantly `/users/team_table` 

Move `get 'users/team_table'` to above the `resources :users ...` routes in `routes.rb` and you'll get a new error about needing to define a new `team_table` action on `UsersController`.

**Pull request merged**